lib/src/native/realm_core.dart

```diff
@@ -764,8 +764,35 @@ class _RealmCore {
     _realmLib.invokeGetBool(() => _realmLib.realm_rollback(realm.handle._pointer), "Could not rollback write");
   }
 
-  void realmRefresh(Realm realm) {
+  bool realmRefresh(Realm realm) {
     _realmLib.invokeGetBool(() => _realmLib.realm_refresh(realm.handle._pointer), "Could not refresh");
+    //TODO: Remove this and return the correct result of calling `realm_refresh`.
+    //Currently there is a bug in C-API that does not return the refresh result correctly.
+    //Issue https://github.com/realm/realm-core/pull/6068
+    return true;
+  }
+  
+  Future<void> realmRefreshAsync(Realm realm) {
```

Comment:
Maybe not for this PR, as this is a pattern we have used all over `realm_core.dart`..

Anyway.. Why not make this 
```suggestion
  Future<void> realmRefreshAsync(Realm realm) async {
```
and await the `completer.future` at the end. That way we are sure that any exceptions raised by this functions before that point is correctly wrapped in a `Future.error(ex)`

Reply:
I was debating to the same but for the public API which is the entry point. so `Realm.refreshAsync`. Let's discuss that and we can do it for all. But not in this PR :)